Katello/katello
# Fixes #34792 - Do not show duplicate module streams in the new UI

#### What are the changes introduced in this pull request?

* Added a new method to filter out duplicate streams and only return one of each for the new ui.

#### Considerations taken when implementing this change?

* Made sure Module stream applicabilty was not affected

#### What are the testing steps for this pull request?

* Apply patch
* Sync CentOS8 stream repo from anywhere or this mirror (https://mirror.steadfast.net/centos/8-stream/BaseOS/x86_64/os/) - I synced RHEL 8 BaseOS/AppStream and registered a RHEL 8 host but it should work with CentOS
* Register a CentOS 8 stream machine to your Katello server
* Verify you do not see duplicate module streams in the new host UI

![screenshot-192 168 121 231-2022 07 18-12_11_38](https://user-images.githubusercontent.com/1518655/179556000-554c64bf-585b-4ef0-938e-98e8b8895293.png)
